internal/controllers/topology/cluster/desired_state.go

```diff
@@ -81,18 +81,12 @@ func (r *Reconciler) computeDesiredState(ctx context.Context, s *scope.Scope) (*
 	// - Building the TopologyReconciled condition.
 	// - Make upgrade decisions on the control plane.
 	// - Making upgrade decisions on machine pools.
-	if len(s.Current.MachinePools) > 0 {
-		client, err := r.Tracker.GetClient(ctx, client.ObjectKeyFromObject(s.Current.Cluster))
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to check if any MachinePool is upgrading")
-		}
-		// Mark all the MachinePools that are currently upgrading.
-		mpUpgradingNames, err := s.Current.MachinePools.Upgrading(ctx, client)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to check if any MachinePool is upgrading")
-		}
-		s.UpgradeTracker.MachinePools.MarkUpgrading(mpUpgradingNames...)
+	// Mark all the MachinePools that are currently upgrading.
+	mpUpgradingNames, err := s.Current.MachinePools.Upgrading(ctx, r.Client)
```

Comment:
We can't use r.Client here, because we want to get the nodes from the remote cluster. This client would try to get the nodes from the management cluster.